be/src/olap/column_predicate.h

```diff
@@ -210,6 +212,11 @@ class ColumnPredicate {
 
     virtual bool can_do_bloom_filter(bool ngram) const { return false; }
 
+    // Check input type could apply safely.
+    // Note: Currenly ColumnPredicate is not include complex type, so use PrimitiveType
+    // is simple and intuitive
+    virtual bool can_do_apply_safely(PrimitiveType input_type, bool is_null) const = 0;
```

Comment:
We can define a base impl for can_do_apply_safely since it's the same for most subclass predicate.

Reply:
we need type for comparison， but base impl could not perceive type info，or trick to implement